### ISSUE TITLE
Some details to claify

### ISSUE THREAD

**Issue by yucc-leon** [ASTRAL-Group/data-efficient-llm-rl#1]
Hi, great work with a lot of insights! There are some detailed implementations I haven't quite understood:

1. In `run_bash/run_train.sh` and `train.py`, the argument `freeze_encoder` seems not to be working for GPT-like models having no encoder layer. But the paper said you froze the main weights of the embedding model...
2. What does it mean to train the difficulty proxy/embedding model with generations with rewards across models (like mathtral/qwen/...)?
3. According to the paper and files in `adaptive_difficulty_prediction`, before `run_train.sh`, there should be cross-model generations on a specific dataset. Does the `reward` field mean the pass rate (correct answers from all generations) of this problem?

Please give me some more suggestions to help with the reproduction. Thanks:)

**Comment by jy-evangeline** [comment]
Thanks for your interest!

> In run_bash/run_train.sh and train.py, the argument freeze_encoder seems not to be working for GPT-like models having no encoder layer. But the paper said you froze the main weights of the embedding model...

Yes. In our setup, we ```set use_embedding=True```, so the encoder was not invoked during training — input embeddings were precomputed and cached. While this was not implemented via an explicit freeze, it is functionally equivalent: the backbone model’s parameters were not updated during training. We have also updated the code to support explicit freezing for decoder-only architectures such as Qwen.

> What does it mean to train the difficulty proxy/embedding model with generations with rewards across models (like mathtral/qwen/...)? According to the paper and files in adaptive_difficulty_prediction, before run_train.sh, there should be cross-model generations on a specific dataset. Does the reward field mean the pass rate (correct answers from all generations) of this problem?

As stated in our paper, to train the adaptive prediction framework, we collect training data from a set of LLMs that are disjoint from our policy models.  For **each** model, we sample query questions and reference questions from math datasets and compute their adaptive difficulty as supervision labels. In the codebase, this is stored in the ```reward``` field, which corresponds to the negative adaptive difficulty. Specifically, adaptive difficulty is defined as the average failure rate of a given question under the current model across rollouts. Each training instance consists of:

- a query question `q`
- a reference set `{(q_i, d_i)}_{i=1}^K` with known difficulty scores
- a ground-truth difficulty label `d_q`

Repeating this procedure across models yields the training dataset.
